### ISSUE TITLE
Figure 6.5 error

### ISSUE THREAD

**Issue by kdhiner** [astroML/astroML#56]
Got an error while plotting figure 6.5:
    ax.plot(t, true_pdf(t), ':', color='black', zorder=3,
            label="Generating Distribution")

The error stated that x and y must have the same dimension. I looked and found that true_pdf returns a single value, while the array t had 1000 elements. 

I corrected this myself by adding the following code after defining t:

    t_true = []
    for item in t:
        t_true.append(true_pdf(item))

and then plotting t_true instead of true_pdf(t). I don't know if that is the most efficient way of correcting the error, but it worked and produced the figure as it appears on the website.

**Comment by jakevdp** [comment]
Did you run the script as-is, or did you do it in an environment with other code? If you executed ``from numpy import *`` or, in IPython ran ``%pylab`` before running this script, then that would over-write the built-in ``sum`` function and lead to this error.

I'd recommend never using ``%pylab`` or importing * from packages due to issues like this.

**Comment by kdhiner** [comment]
Thanks for the information. I have been running in ipython notebook with pylab, so that's probably why I got the error initially. I just ran the code without pylab enabled and it worked as-is. 

So the numpy sum function sums over all dimensions by default. Instead of inserting the code above as I did, I could also have specified 'axis=0' in the sum function (with pylab enabled). Without pylab, it works as-is. 

I understand now. Thanks.
